src/resource.ts

```diff
@@ -0,0 +1,346 @@
+   */
+  public withParent(parent: IParentParams): this {
+    if (parent && parent.id) {
+      this.setParentId(parent.id);
+    }
+    return this;
+  }
+
+  /**
+   * Create a resource by ID
+   *
+   * @params params - Resource-specific parameters
+   * @params cb - (DEPRECATED SINCE 2.2.0) Callback function, can be used instead of the returned Promise
+   *
+   * @returns The resource
+   *
+   * @since 1.0.0
+   *
+   * @public ✓ This method is part of the public API
+   */
+  public async create(prms: any, cb?: ResourceCallback): Promise<Model> {
+    const params = cloneDeep(prms);
+    const callback = typeof params === 'function' ? params : cb;
+    const query: any = {};
+    if (isPlainObject(params)) {
+      if (typeof params.include === 'string' || Array.isArray(params.include)) {
+        query.include = Array.isArray(params.include) ? params.include.join(';') : params.include;
+        delete params.include;
+      }
+      if (typeof params.embed === 'string' || Array.isArray(params.embed)) {
+        query.embed = Array.isArray(params.embed) ? params.embed.join(';') : params.embed;
+        delete params.embed;
+      }
+    }
+
+    try {
+      const response: AxiosResponse = await this.getClient().post(`${this.getResourceUrl()}${qs.stringify(query, { addQueryPrefix: true })}`, params);
+
+      // noinspection JSPotentiallyInvalidConstructorUsage
+      const model = new (this.constructor as any).model(response.data);
+
+      if (callback) {
+        return callback(null, model);
+      }
+      return model;
+    } catch (error) {
+      Resource.errorHandler(error.response, callback);
+    }
+  }
+
+  /**
+   * Get a resource by ID
+   *
+   * @params id - Resource ID
+   * @params params - Resource-specific parameters
+   * @params cb - (DEPRECATED SINCE 2.2.0) Optional callback function
+   *
+   * @returns {Promise<Model>}
+   *
+   * @since 1.0.0
+   */
+  public async get(id: string, prms?: any, cb?: ResourceCallback): Promise<Model> {
+    const params = cloneDeep(prms);
+    const callback = typeof params === 'function' ? params : cb;
+    const query: any = {};
+    if (isPlainObject(params)) {
+      if (typeof params.include === 'string' || Array.isArray(params.include)) {
+        query.include = Array.isArray(params.include) ? params.include.join(';') : params.include;
+        delete params.include;
+      }
+      if (typeof params.embed === 'string' || Array.isArray(params.embed)) {
+        query.embed = Array.isArray(params.embed) ? params.embed.join(';') : params.embed;
+        delete params.embed;
+      }
+    }
+
+    try {
+      const response: AxiosResponse = await this.getClient().get(`${this.getResourceUrl()}/${id}${qs.stringify(query, { addQueryPrefix: true })}`);
+
+      // noinspection JSPotentiallyInvalidConstructorUsage
+      const model = new (this.constructor as any).model(response.data);
+
+      if (callback) {
+        return callback(null, model);
```

Comment:
Hmmz, that should be double-checked, indeed. I believe my IDE complained about not returning otherwise, because by marking a function as `async` we are at leasting promising to return a `Promise`.